sklearn_hierarchical_classification/classifier.py

```diff
@@ -275,29 +275,44 @@ def _recursive_build_features(self, X, y, node_id, progress):
                 y=y,
                 indices=indices,
             )
+
             return self.graph_.nodes[node_id]["X"]
 
         # Non-leaf node
         self.graph_.nodes[node_id]["X"] = csr_matrix(
             X.shape,
             dtype=X.dtype,
         )
+
         for child_node_id in self.graph_.successors(node_id):
             self.graph_.nodes[node_id]["X"] += \
                 self._recursive_build_features(
                     X=X,
                     y=y,
                     node_id=child_node_id,
                     progress=progress,
-            )
+                )
 
         # Build and store meta-features for node
         self.graph_.nodes[node_id][METAFEATURES] = self._build_metafeatures(
             X=self.graph_.nodes[node_id]["X"],
             y=y,
         )
 
-        return self.graph_.nodes[node_id]["X"]
+        # Append training data tagged with current (intermediate) node if any, and propagate up
+        if not np.issubdtype(type(node_id), y.dtype):
+            # If current intermediate node id type is different than that of targets array, dont bother.
+            # Nb. doing this check explicitly to avoid FutureWarning, see:
+            # https://stackoverflow.com/questions/40659212/futurewarning-elementwise-comparison-failed-returning-scalar-but-in-the-futur
+            return self.graph_.nodes[node_id]["X"]
+
+        indices = np.flatnonzero(y == node_id)
+        X_out = self.graph_.nodes[node_id]["X"] + self._build_features(
+            X=X,
+            y=y,
+            indices=indices,
+        )
```

Comment:
^ this is the crux of the PR.

When we recursively build training data for each node, we "bubble up" labels from child nodes. we need to also add training data labeled with the current node though for cases where it shows up in the training data, which we didnt do before.